test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/fields/leaf/ByteFieldDataGenerator.java

```diff
@@ -54,4 +54,9 @@ public CheckedConsumer<XContentBuilder, IOException> mappingWriter() {
     public CheckedConsumer<XContentBuilder, IOException> fieldValueGenerator() {
         return b -> b.value(valueGenerator.get());
     }
+
+    @Override
+    public Object generateValue() {
+        return valueGenerator.get();
+    }
```

Comment:
These three functions seem to repeat.. maybe just define them in `FieldDataGenerator` and pass the field type name.